### ISSUE TITLE
Limiting file size

### ISSUE THREAD

**Issue by ChrisButterworth** [yabwe/medium-editor#1180]
Looking for a way to limit the size of the images that can be dragged into the editor

Tried subscribing to the `editableDrag` and `editableDrop` events to no avail as well as using `.on` drop event.

Cannot figure this out so any guidance would be great.

Using the latest version (5.21.1) and everything is else running great.

**Comment by j0k3r** [comment]
Could you provide an example?
This example is working fine for me when you drag'n'drop an image: https://jsfiddle.net/h0m6767q/
Check the console.

**Comment by ChrisButterworth** [comment]
Didn't explain myself, I can tap into those events but can't stop the image being added when you drag and drop it.

Example retuning false in an attempt to stop the image being dropped.
https://jsfiddle.net/jusuj2bp/

**Comment by j0k3r** [comment]
(your jsfiddle is exactly the same as mine, did you forget to update it?)
Anyway, I get your point.

It seems that you can't disable drop by simply subscribing to the event `editableDrop`.

**Comment by ChrisButterworth** [comment]
Such a noob move....
https://jsfiddle.net/jusuj2bp/1/

Can use both
`event.dataTransfer.effectAllowed = "none";`
`event.dataTransfer.dropEffect = "none";`
on the `editableDrag` event

But can't access `DataTranser.files` to do any form of validation

**Comment by j0k3r** [comment]
I've made some tests: https://jsfiddle.net/jusuj2bp/5/
It seems that the files entry is updated when the file dropped and not when it's dragged, _at least on Chrome 52_.

**Comment by ChrisButterworth** [comment]
You can get what I'm looking to achieve from this one.

https://jsfiddle.net/jusuj2bp/7/

Looking for a way to stop the image being added.

**Comment by j0k3r** [comment]
I got your point.

**Comment by ChrisButterworth** [comment]
Excellent, I thought it would have been fairly simple to do.

**Comment by nmielnik** [comment]
Yeah, I believe there have been requests for similar things on some of our other events.

So the first thing we'd need to do have some mechanism for the application to 'preventDefault' on the event.  Right now, we just tigger events but don't have any way to know if any of the listeners did anything.  Maybe we would expose a `preventDefault()` method that the internal code can know when it was called.

The next thing would be coming up with new event names for these.  I believe most of our events are triggered after something has happened, so it would be too late to do anything about it.  We wouldn't want to change the time when events are fired since this could definitely break folks using these events.

So, we could have something like a new `beforeFileInserted` event, which would get triggered before the editor actually tried to insert it, and would be able to cancel the action if any listeners to the event specified they want it canceled (ie `preventDefault()`).

This could be useful in many ways beyond just this use case, so I'm definitely on board with this enhancement being added.

**Comment by Jarown** [comment]
I ran into the same problem. Since it seems there's no built-in support for pre-validation (yet) I used this approach/workaround to prevent images being dropped if they are invalid (too big, unsupported file-type, ..): https://jsfiddle.net/jarown/ku106ng7/

It basically comes down to disabling the built-in image dragging with `imageDragging: false` and take care of processing, checking and inserting the files yourself.
